feature_engine/selection/drop_duplicate_features.py

```diff
@@ -0,0 +1,76 @@
+import itertools
+from sklearn.base import TransformerMixin, BaseEstimator
+from sklearn.utils.validation import check_is_fitted
+from feature_engine.dataframe_checks import _is_dataframe, _check_input_matches_training_df
+
+
+class DropDuplicateFeatures(BaseEstimator, TransformerMixin):
+    """
+    Drop duplicate features from a dataframe. Duplicate features are those set of features which show same value across
+    all observations.
+    """
+
+    def __init__(self):
+        pass
+
+    def fit(self, X, y=None):
+
+        """
+        Find duplicated features.
+
+        Parameters
+        ----------
+
+        X: pandas dataframe of shape = [n_samples, n_features]
+            The input dataframe.
+
+        y: None
+            y is not needed for this transformer. You can pass y or None.
+
+
+        Attributes
+        ----------
+
+        duplicate_feature_dict_: dict
+            The dictionary where keys are features and values are duplicate features.
+        """
+
+        # check input dataframe
+        X = _is_dataframe(X)
+
+        duplicated_features_list = []
+
+        # create column pairs
+        col_pair = list(itertools.combinations(X.columns, 2))
+
+        # create a dictionary of features with values as list of duplicate features
+        self.duplicate_feature_dict_ = {col: [] for col in X.columns}
+
```

Comment:
Hi @solegalli, thanks for the comments. My intention was to choose dict so user knows what features are found duplicate to other features and what we keep(```dict.keys()```). If we just output duplicates another question could arise, those duplicates are to which features? For example, in this implementation ```self.duplicated_features_ = [ [ a,b,c], [d,e,f], [h,j,k] ]``` inner list would consist duplicate features like a,b,c which one we keep, which one we drop? Also, will list of list is going to show that if it is found duplicate to ```X``` feature? If it somehow shows that feature ```X``` is duplicate to  features```[a,b,c]``` it will help me gain confidence in the data.  Thoughts?

Reply:
Or maybe we just output duplicates in the form of a dictionary and internally play with an array?